libs/ngx-gauge/src/gauge/gauge.ts

```diff
@@ -256,21 +256,20 @@ export class NgxGauge implements AfterViewInit, OnChanges, OnDestroy {
             displacement = unit * nv - unit * ov;
         }
         function animate(timestamp) {
+            const requestID = requestAnimationFrame(animate);
             timestamp = timestamp || new Date().getTime();
             let runtime = timestamp - startTime;
             let progress = Math.min(runtime / duration, 1);
             let previousProgress = ov ? ov * unit : 0;
             let middle = start + previousProgress + displacement * progress;
 
             self._drawShell(start, middle, tail, color);
-            if (self._animationRequestID && (runtime < duration)) {
-                self._animationRequestID = window.requestAnimationFrame((timestamp) => animate(timestamp));
-            } else {
-                window.cancelAnimationFrame(self._animationRequestID);
+            if (progress === 1) {
+                cancelAnimationFrame(requestID);
```

Comment:
By calling cancelAnimationFrame, this results in a series of undefined requests in the `.removeRect` function. I tested the code again and if you keep  the original if condition
`if (self._animationRequestID && (runtime < duration)) {
self._animationRequestID = window.requestAnimationFrame((timestamp) => animate(timestamp));`

but remove the `else` statement, then it still is able to draw but does not generate a slew of errors in the console.